src/ray/core_worker/task_manager.h

```diff
@@ -304,10 +331,10 @@ class TaskManager : public TaskFinisherInterface, public TaskResubmissionInterfa
     // Number of times this task may be resubmitted. If this reaches 0, then
     // the task entry may be erased.
     int num_retries_left;
+    // Reference to the task stats tracker.
+    TaskStatusCounter &counter;
```

Comment:
Is there a way to check if lock is acquired? It looks like at the top level (task_manager.h), it is guarded by the mutex, but when it is used inside the `TaskEntry`, it is not required. (is lock required when TaskEntry is accessed?)

Reply:
Not that I know of without passing in the mutex pointer (not sure if it's worth it).

Reply:
My personal preference is to just pass it and lock it, but I think it is up to you. (I assume the `TaskEntry ` is always protected by the mutex now).